wordpress-mobile/WordPress-Android
# Add custom theme for Design System

Part of issue https://github.com/wordpress-mobile/WordPress-Android/issues/19802

 This PR adds the `DesignSystemTheme` Compose theme with a custom `DesignSystemColors` palette. `DesignSystemColors` should contain all attributes defined in the Figma spec [here](https://www.figma.com/file/LyzJJa6ANKToUswWmRrZee/%F0%9F%A7%B1-Jetpack-App-Design-System-iOS?type=design&node-id=102-186&mode=design&t=3p94G4176bfbDwns-0).

-----

## To Test:
1. Navigate to the `Me` tab. If you're on a debug build, you should see the `Design System` row.
2. Toggle between `light mode`, `dark mode` and `system default` under `App Settings`. 
3. With each of the modes above, navigate to the `Design System` and each of the screens should be themed accordingly.

Dark mode:
<img width="405" alt="Screenshot 2024-01-30 at 4 05 50 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/7199140/1c8c115b-33c3-4f3e-afa9-630bfdd8ba44">

Light mode: 
<img width="407" alt="Screenshot 2024-01-30 at 4 05 43 PM" src="https://github.com/wordpress-mobile/WordPress-Android/assets/7199140/e2a5faac-69de-4395-a342-dd9c195aaeaf">




-----

## Regression Notes

1. Potential unintended areas of impact

    - None

5. What I did to test those areas of impact (or what existing automated tests I relied on)

    - None

6. What automated tests I added (or what prevented me from doing so)

    - None

-----

## PR Submission Checklist:


-----

## UI Changes Testing Checklist:
